### PR TITLE
Update local_network script for new docker target dir

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously the local_network script looked in `target/release` for the
binaries.  Now the local_network script will attempt to use the
`CARGO_TARGET_DIR` if it's set.

Also updates associated markdown files so copy and paste of commands
works in docker containers.

### Motivation
The local_network script no longer works without local modification when run from inside of the docker image via `./mob prompt`

### Future Work

[Soundtrack of this PR]()

### CODE REVIEW

File: consensus/service/README.md [hunk 1]
```diff
@@ -173,7 +173,7 @@ SGX_MODE=HW IAS_MODE=DEV \
 Alternatively, if the binary has already been built, you can run:
 
 ```
-./target/release/consensus-service --client-responder-id \
+${CARGO_TARGET_DIR}/release/consensus-service --client-responder-id \
```
Comment (remoun) [inline consensus/service/README.md:176]:
```suggestion
${CARGO_TARGET_DIR:-target}/release/consensus-service --client-responder-id \
```

File: tools/fog-local-network/README.md [hunk 2]
```diff
@@ -58,7 +58,8 @@ In order to use it, the following steps are necessary.
     # Create a set of target keys. They would be identical to the first N keys inside `keys/`. This is needed if you don't
     # want to send to transactions to all 1000 keys created at step 1.
     # Notice the addition of the --output-dir argument
-    cargo run -p mc-util-keyfile --bin sample-keys --release -- --num 10 --output-dir fog_keys --fog-report-url 'insecure-fog://localhost:6200' --fog-authority-root $(../target/release/mc-crypto-x509-test-vectors --type=chain --test-name=ok_rsa_head)
+    cargo run -p mc-util-keyfile --bin sample-keys --release -- --num 10
+    --output-dir fog_keys --fog-report-url 'insecure-fog://localhost:6200' --fog-authority-root $(${CARGO_TARGET_DIR:-target}/release/mc-crypto-x509-test-vectors --type=chain --test-name=ok_rsa_head)
```
Comment (remoun) [inline tools/fog-local-network/README.md:62]: Remove the line break, or add `\` before it

```suggestion
    AUTHORITY_ROOT=$(${CARGO_TARGET_DIR:-target}/release/mc-crypto-x509-test-vectors --type=chain --test-name=ok_rsa_head)
    cargo run -p mc-util-keyfile --bin sample-keys --release --
        --num 10 \
        --output-dir fog_keys \
        --fog-report-url 'insecure-fog://localhost:6200' \
        --fog-authority-root $AUTHORITY_ROOT
```

File: tools/local-network/local_network.py [hunk 3]
```diff
@@ -30,14 +30,14 @@
 PROJECT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
 MOB_RELEASE = os.getenv('MOB_RELEASE', '1')
 CARGO_FLAGS = '--release'
-TARGET_DIR = 'target/release'
-WORK_DIR =  os.path.join(PROJECT_DIR, TARGET_DIR, 'mc-local-network')
+TARGET_DIR = os.path.join(os.getenv('CARGO_TARGET_DIR', os.path.join(PROJECT_DIR, 'target')), 'release')
+WORK_DIR =  os.path.join(TARGET_DIR, 'mc-local-network')
```
Comment (remoun) [inline tools/local-network/local_network.py:34]: Optional/Equivalent: Extract a var for the base target dir, and use conditional expressions to replace the `if MOB_RELEASE == '0':` block below.

```suggestion
MOB_RELEASE = os.getenv('MOB_RELEASE', '1') == '1'
CARGO_FLAGS = '--release' if MOB_RELEASE else ''
BASE_TARGET_DIR = os.getenv('CARGO_TARGET_DIR', os.path.join(PROJECT_DIR, 'target'))
TARGET_DIR = os.path.join(BASE_TARGET_DIR, 'release' if MOB_RELEASE else 'debug')
```

**Review by remoun — APPROVED** [review]

File: consensus/service/README.md (hunk 1 above)
Comment (nick-mobilecoin) [reply consensus/service/README.md]: Should be addressed now, the suggestion was slightly modified to accomodate for `./target`

File: tools/fog-local-network/README.md (hunk 2 above)
Comment (nick-mobilecoin) [reply tools/fog-local-network/README.md]: pushed up a resolution in newest commit

File: tools/local-network/local_network.py (hunk 3 above)
Comment (nick-mobilecoin) [reply tools/local-network/local_network.py]: Did a similar and slightly different approach, the `if MOB_RELEASE` will always evaluate to true
```
>>> foo = "0"
>>> if foo: print("Toasty!!!")
... 
Toasty!!!
```

Comment (remoun) [reply tools/local-network/local_network.py]: Note that my suggestion changes the definition to be a proper boolean

**Review by eranrund — COMMENTED** [review]
I'm curious what surfaced this. When does  `CARGO_TARGET_DIR` gets set?

**Comment by nick-mobilecoin** [comment]
> I'm curious what surfaced this. When does `CARGO_TARGET_DIR` gets set?

https://github.com/mobilecoinfoundation/mobilecoin/pull/1935/files brought in a change that sets the `CARGO_TARGET_DIR` for the docker invocation.

**Review by eranrund — APPROVED** [review]
LGTM once CI is happy

**Review by iamalwaysuncomfortable — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
@eranrund and @iamalwaysuncomfortable you may want to give another look over now that I figured out the CI error, https://github.com/mobilecoinfoundation/mobilecoin/pull/1961/commits/966e72815f1ab13385682c85d72e043dcbcf3fb2

File: tools/local-network/local_network.py [hunk 4]
```diff
@@ -29,16 +28,19 @@
...
-TARGET_DIR = 'target/release'
+if MOB_RELEASE == '1':
+    CARGO_FLAGS = '--release'
+    BUILD_TYPE = 'release'
+else:
+    CARGO_FLAGS = ''
+    BUILD_TYPE = 'debug'
+
+BASE_TARGET_DIR = os.getenv('CARGO_TARGET_DIR', 'target')
+TARGET_DIR = os.path.join(os.path.relpath(BASE_TARGET_DIR, PROJECT_DIR), BUILD_TYPE)
```
Comment (eranrund) [inline tools/local-network/local_network.py:39]: I think going back and forth from relative to absolute paths is confusing.
How about something like this?
```
PROJECT_TARGET_DIR = os.path.join(PROJECT_DIR, 'target') # The default target directory
BASE_TARGET_DIR = os.path.abspath(os.getenv('CARGO_TARGET_DIR', PROJECT_TARGET_DIR)) # Allow overriding with the `CARGO_TARGET_DIR` env var
TARGET_DIR = os.path.join(BASE_TARGET_DIR, BUILD_TYPE) # Append `debug`/`release`
WORK_DIR = os.path.join(TARGET_DIR, 'mc-local-network')
```

**Review by remoun — APPROVED** [review]
LGTM

File: tools/local-network/local_network.py (hunk 4 above)
Comment (nick-mobilecoin) [reply tools/local-network/local_network.py]: Should be similar

**Comment by iamalwaysuncomfortable** [comment]
@nick-mobilecoin - this okay to merge?

**Comment by nick-mobilecoin** [comment]
Yes, sorry its in now

**Pull request merged**